src/psyclone/gocean1p0.py

```diff
@@ -1332,13 +1338,14 @@ def gen_ocl(self, parent):
                             qlist, str(previous_queue))))
 
         # If the dependency is something other than a kernel, currently we
-        # dispatch everything else to queue 1, so add a barrier if this kernel
-        # is not on queue number 1.
+        # dispatch everything else to queue _OCL_MANAGEMENT_QUEUE, so add a
+        # barrier if this kernel is not on queue _OCL_MANAGEMENT_QUEUE.
         if dependency and not dependency.coded_kernels() and queue_number != 1:
```

Comment:
Should be `queue_number != _OCL_MANAGEMENT_QUEUE` I think?

Reply:
Yes, updated